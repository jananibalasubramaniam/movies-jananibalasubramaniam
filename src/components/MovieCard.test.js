import { render } from '@testing-library/react';
import MovieCard from './MovieCard';
import '@testing-library/jest-dom/extend-expect';

describe('The MovieCard component', () => {
  it('should render movie title and release date', () => {
    const { container } = render(
      <MovieCard movie={{
        title: 'I am a test movie',
        poster: 'https://image.tmdb.org/t/p/w600_and_h900_bestv2/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg',
        released: new Date('March 24, 2021'),
      }} />, {},
    );
    expect(container).toHaveTextContent('I am a test movie');
    expect(container).toHaveTextContent('March 24, 2021');
    expect(container.firstChild).toMatchSnapshot();
  });
});
