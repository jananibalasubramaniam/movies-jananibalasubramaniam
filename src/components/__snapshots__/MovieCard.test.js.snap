// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MovieCard component should render movie title and release date 1`] = `
<div
  class="MuiPaper-root MuiCard-root Component-backdropCard-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root"
    tabindex="0"
    type="button"
  >
    <img
      alt="I am a test movie"
      class="MuiCardMedia-root Component-media-3 MuiCardMedia-media MuiCardMedia-img"
      title="I am a test movie"
    />
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6"
      >
        I am a test movie
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        March 24, 2021
      </p>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
